### PR TITLE
impl(otel): conversion to `google.api.Metric` helper

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #13869 

`time_series.h` will ultimately contain helpers that convert from the OpenTelemetry SDK to `google.monitoring.v3.TimeSeries` / `google.monitoring.v3.CreateTimeSeriesRequest`. We start with a helper for `google.api.Metric`

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13906)

### CODE REVIEW

File: google/cloud/opentelemetry/internal/time_series.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_OPENTELEMETRY_INTERNAL_TIME_SERIES_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_OPENTELEMETRY_INTERNAL_TIME_SERIES_H
+
+#include "google/cloud/version.h"
+#include <google/monitoring/v3/metric_service.pb.h>
```
Comment (devbww) [inline google/cloud/opentelemetry/internal/time_series.h:19]: Unused?

File: google/cloud/opentelemetry/internal/time_series.cc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/opentelemetry/internal/time_series.h"
+#include "google/cloud/opentelemetry/internal/monitored_resource.h"
+#include "google/cloud/internal/absl_str_replace_quiet.h"
+#include "absl/types/variant.h"
```
Comment (devbww) [inline google/cloud/opentelemetry/internal/time_series.cc:18]: Unused?

File: google/cloud/opentelemetry/internal/time_series.cc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+namespace cloud {
+namespace otel_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+google::api::Metric ToMetric(
+    opentelemetry::sdk::metrics::MetricData const& metric_data,
+    opentelemetry::sdk::metrics::PointAttributes const& attributes) {
+  google::api::Metric proto;
+  auto const metric_type =
+      "workload.googleapis.com/" + metric_data.instrument_descriptor.name_;
```
Comment (coryan) [inline google/cloud/opentelemetry/internal/time_series.cc:31]: This hardcoded prefix is suspect.  And I believe maybe it does not work for custom metrics:

https://cloud.google.com/monitoring/api/v3/naming-conventions

That suggests all metrics created by the application go to `custom.googleapis.com`.

**Review by coryan — APPROVED** [review]
Is that prefix right?  If it worked for you then feel free to merge.

File: google/cloud/opentelemetry/internal/time_series.cc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  proto.set_type(metric_type);
+
+  for (auto const& kv : attributes) {
+    auto& labels = *proto.mutable_labels();
+    // GCM labels match on the regex: R"([a-zA-Z_][a-zA-Z0-9_]*)". We could
+    // convert any non-matching character to '_', but let's just do some usual
```
Comment (devbww) [inline google/cloud/opentelemetry/internal/time_series.cc:37]: Any reason not to do exactly that?

**Review by devbww — APPROVED** [review]

File: google/cloud/opentelemetry/internal/time_series.cc (hunk 3 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/internal/time_series.cc:31]: The other exporters use `workload.googleapis.com`. That is what is in: go/otel-gcp-metric-exporter-spec

> hardcoded

It is configurable in other exporters. I am not there yet.

> maybe it does not work for custom metrics

It does work. I have tried it. https://screenshot.googleplex.com/94MR7j6wFz7frnA

File: google/cloud/opentelemetry/internal/time_series.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/internal/time_series.h]: It was used, but indirectly

File: google/cloud/opentelemetry/internal/time_series.cc (hunk 2 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/internal/time_series.cc]: Removed.

File: google/cloud/opentelemetry/internal/time_series.cc (hunk 4 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/internal/time_series.cc]: Oops, I messed this up. I will need another commit.

Comment (dbolduc) [reply google/cloud/opentelemetry/internal/time_series.cc]: Ok, fixed. (I didn't notice the label can start with an underscore).

**Pull request merged**